pkg/agent/flexvolume/controller.go

```diff
@@ -0,0 +1,341 @@
+/*
+Copyright 2017 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+Some of the code below came from https://github.com/coreos/etcd-operator
+which also has the apache 2.0 license.
+*/
+
+// Package flexvolume to manage Kubernetes storage attach events.
+package flexvolume
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"path"
+	"strings"
+
+	"github.com/coreos/pkg/capnslog"
+	"github.com/rook/rook/pkg/agent/flexvolume/crd"
```

Comment:
why split the `crd` and `manager/ceph' subdirs? Seems simpler to have them in the same package

Reply:
my reasoning is that `crd` is about managing the Volumeattachment CRD, which is independent with the backend technology that is doing the mounting. So in the future when we add support for gluster, the CRD controller is independent.

Reply:
if we add gluster, we would use the same VolumeAttachment crd and have a `manager/gluster` dir, right? Makes sense. I think `crd` is an awkward package name though. What if you called it `attach` or something more descriptive like that?

Reply:
correct. the `manager` package contains all the backend specific code to perform the attachment.

`attach` doesnt say that everything under this package is about registering, managing and controlling crd.
I tried naming it volumeattachmentcrd but seems long and wordy.

Reply:
package names are supposed to be very short and they have the context in the full path. Is there another single word besides `attach` that makes sense? Because `crd` is very generic

Reply:
`crd` is generic but since it is a sub package of `flexvolume` i think it is ok. I could rename it `attachcrd` but i still like `crd` better